### ISSUE TITLE
Permalinks for Assemble v0.6.0

### ISSUE THREAD

**Issue by Melindrea** [assemble/assemble-middleware-permalinks#60]
How would I go about using this middleware (or get the same effect) in v0.6? I'm not needing anything too advanced, just standard 'pretty links' and some subfolders. That is:

```
/pages
    -- index => /
    -- colophone => /colophone
    /stories
    -- index => /stories
    -- story-1 => /stories/story-1
```

**Comment by jonschlinkert** [comment]
Coincidentally I just started working on updating this for assemble v0.6.0. I'll update when it's ready

**Comment by Melindrea** [comment]
Thanks, sounds good =)

I'll work on other things, and if the permalinks aren't done in time for my next version, I should hopefully be able to juryrig an ugly placeholder!

**Comment by jonschlinkert** [comment]
lol, I know the feeling. I'm in a similar position so I want to get this done!

**Comment by Melindrea** [comment]
If there's any particular bit I can help with, let me know

**Comment by jonschlinkert** [comment]
Awesome thanks. Any feedback on what features you want to see would be great. I haven't decide how much I'm going to stick to the original feature set

**Comment by Melindrea** [comment]
I'll think over that some. I managed to get the most naive implementation possible for my current set of pages, but my real website will also need a blog with categories/tags/etc.

Naive implementation:

```
.pipe(rename(function (path) {

            if (path.basename !== 'index') {
                path.dirname += '/' + path.basename;
                path.basename = 'index';
            } else {
                path.dirname = '.';
            }
        }))
```

Aka, if it's index it is in the same folder, if not it gets moved to basename/index.html

I'm not renaming the extension, since I use gulp-extname (which I kind-of figure I'll want to keep later as well).

**Comment by jonschlinkert** [comment]
Thanks, that helps!

**Comment by Melindrea** [comment]
And just confirmed that it works with at least three levels of subfolders, and with a page called `stories.md` alongside the folder `stories`, nesting it properly.

**Comment by Melindrea** [comment]
>.< Okay, so disregard this ATM. It printed out the paths correctly ... but it apparently didn't like the idea of giving them the right content!

**Comment by jonschlinkert** [comment]
being implemented as a middleware in 0.6
